plugins/inputs/ping/ping_windows.go

```diff
@@ -83,61 +95,80 @@ func (p *Ping) args(url string) []string {
-// based on linux implementation but using regex ( multilanguage support )
+// based on linux implementation but using regex (multi-language support)
 // It returns (<transmitted packets>, <received reply>, <received packet>, <average response>, <min response>, <max response>)
-func processPingOutput(out string) (int, int, int, int, int, int, error) {
+func processPingOutput(out string) (statistics, error) {
 	// So find a line contain 3 numbers except reply lines
-	var stats, aproxs []string = nil, nil
+	var statsLine, aproxs []string = nil, nil
 	err := errors.New("fatal error processing ping output")
 	stat := regexp.MustCompile(`=\W*(\d+)\D*=\W*(\d+)\D*=\W*(\d+)`)
 	aprox := regexp.MustCompile(`=\W*(\d+)\D*ms\D*=\W*(\d+)\D*ms\D*=\W*(\d+)\D*ms`)
 	tttLine := regexp.MustCompile(`TTL=\d+`)
 	lines := strings.Split(out, "\n")
-	var receivedReply int = 0
+	var replyReceived = 0
 	for _, line := range lines {
 		if tttLine.MatchString(line) {
-			receivedReply++
+			replyReceived++
 		} else {
-			if stats == nil {
-				stats = stat.FindStringSubmatch(line)
+			if statsLine == nil {
+				statsLine = stat.FindStringSubmatch(line)
 			}
-			if stats != nil && aproxs == nil {
+			if statsLine != nil && aproxs == nil {
 				aproxs = aprox.FindStringSubmatch(line)
 			}
 		}
 	}
 
-	// stats data should contain 4 members: entireExpression + ( Send, Receive, Lost )
-	if len(stats) != 4 {
-		return 0, 0, 0, -1, -1, -1, err
+	stats := statistics{
+		packetsTransmitted: 0,
+		replyReceived:      0,
+		packetsReceived:    0,
+		roundTripTimeStats: roundTripTimeStats{
+			min: -1,
+			avg: -1,
+			max: -1,
+		},
 	}
-	trans, err := strconv.Atoi(stats[1])
+
+	// statsLine data should contain 4 members: entireExpression + ( Send, Receive, Lost )
+	if len(statsLine) != 4 {
+		return stats, err
+	}
+	packetsTransmitted, err := strconv.Atoi(statsLine[1])
 	if err != nil {
-		return 0, 0, 0, -1, -1, -1, err
+		return stats, err
 	}
-	receivedPacket, err := strconv.Atoi(stats[2])
+	packetsReceived, err := strconv.Atoi(statsLine[2])
 	if err != nil {
-		return 0, 0, 0, -1, -1, -1, err
+		return stats, err
 	}
 
+	stats.packetsTransmitted = packetsTransmitted
+	stats.replyReceived = replyReceived
+	stats.packetsReceived = packetsReceived
+
 	// aproxs data should contain 4 members: entireExpression + ( min, max, avg )
 	if len(aproxs) != 4 {
-		return trans, receivedReply, receivedPacket, -1, -1, -1, err
+		return stats, err
 	}
 	min, err := strconv.Atoi(aproxs[1])
 	if err != nil {
-		return trans, receivedReply, receivedPacket, -1, -1, -1, err
+		return stats, err
 	}
 	max, err := strconv.Atoi(aproxs[2])
 	if err != nil {
-		return trans, receivedReply, receivedPacket, -1, -1, -1, err
+		return stats, err
 	}
 	avg, err := strconv.Atoi(aproxs[3])
 	if err != nil {
-		return 0, 0, 0, -1, -1, -1, err
+		return statistics{}, err
 	}
 
-	return trans, receivedReply, receivedPacket, avg, min, max, err
+	stats.avg = avg
+	stats.min = min
+	stats.max = max
```

Comment:
Does this work? Shouldn't it be this?
```suggestion
	stats.roundTripTimeStats.avg = avg
	stats.roundTripTimeStats.min = min
	stats.roundTripTimeStats.max = max
```

Reply:
My version works properly :)

Reply:
Yes, they are called `promoted fields` https://www.geeksforgeeks.org/promoted-fields-in-golang-structure/

Reply:
Thanks, I was confused by these lines:
https://github.com/influxdata/telegraf/pull/13246/files/a2628bd023308c7a22dcbf10c4da74e931a58088#diff-e0d0c1474f3cd04fada4dcff3f889a56ab7cc7e0be4549a7dc2e7651a9b8ec90R126-R130

They made me think `roundTripTimeStats` was a property of `statistics`. 😉